lambda_tiler/handler.py

```diff
@@ -77,19 +78,24 @@ class TilerError(Exception):
     binary_b64encode=True,
 )
 @APP.pass_event
-def viewer(
-    event: object,
-    url: str
-) -> Tuple[str, str, str]:
+def viewer_handler(event: object, url: str, **kwargs: Dict) -> Tuple[str, str, str]:
     """Handle Viewer requests."""
-    endpoint = "https://{domain}/{stage}".format(
-        domain=event['requestContext']['domainName'],
-        stage=event['requestContext']['stage']
-    )
-    html = viewer_template.format(
-        endpoint=endpoint,
-        cogurl=url
-    )
+    host = event["headers"].get("X-Forwarded-Host", event["headers"].get("Host", ""))
+    # Check for API gateway stage
+    if ".execute-api." in host and ".amazonaws.com" in host:
+        stage = event["requestContext"].get("stage", "")
+        host = f"{host}/{stage}"
+
+    scheme = "http://" if host.startswith("127.0.0.1") else "https://"
+    endpoint = f"{scheme}{host}"
+
+    qs = [f"{k}={v}" for k, v in kwargs.items()]
```

Comment:
@bitner I've change the way we determine the endpoint URL. could you double check that it's still working for you ?

Reply:
I won't be able to check today. I'll update and check tomorrow. It looks like should still be fine though.

Reply:
didn't get to checking yesterday. if you break it, i'll pr a fix :-)